### ISSUE TITLE
Regression formset bug introduced from upgrading 3.2.10 to 3.5.10.

### ISSUE THREAD

**Issue by jackton1** [yourlabs/django-autocomplete-light#1135]
BEFORE:

![Fix](https://user-images.githubusercontent.com/17484350/72379649-d3f7df00-36e1-11ea-98cb-1073f92f22b7.gif)

AFTER:

![Jan-14-2020 15-11-40](https://user-images.githubusercontent.com/17484350/72378903-51baeb00-36e0-11ea-9310-09a79f8dc227.gif)

**Comment by VenturaFranklin** [comment]
For me cloning formsets with DAL fields still worked on v3.3.5, upgrading to 3.4.1 killed this and it sill does not work in 3.5.0. So something between 3.3.5 and 3.4.1

Specifically, running the code to allow dynamic add/remove of formsets using jquery: https://medium.com/all-about-django/adding-forms-dynamically-to-a-django-formset-375f1090c2b0

I had to make a few changes to get that tutorial to work, my final cloneMore function is below and this works fine with v3.3.5:
```javascript
function cloneMore(selector, prefix) {
    var newElement = $(selector).clone(true);
    var total = $('#id_' + prefix + '-TOTAL_FORMS').val();
    newElement.find(':input:not([type=button]):not([type=submit]):not([type=reset])').each(function() {
        var old_name = $(this).attr('name');
        if (old_name) {  // This is needed because DAL doesn't have name?
            var name = $(this).attr('name').replace('-' + (total - 1) + '-', '-' + total + '-');
            var id = 'id_' + name;
            $(this).attr({'name': name, 'id': id}).val('').removeAttr('checked');
        }
    });
    newElement.find('label').each(function() {
        var forValue = $(this).attr('for');
        if (forValue) {
          forValue = forValue.replace('-' + (total-1) + '-', '-' + total + '-');
          $(this).attr({'for': forValue});
        }
    });
    newElement.find('span').each(function() {
        // This is needed because DAL will be duplicated otherwise
        $(this).remove();
    });
    total++;
    $('#id_' + prefix + '-TOTAL_FORMS').val(total);
    $(selector).after(newElement);
    updateConditionRow();
    return false;
}
```
The two sections I had to add are noted above. Removing the section below does not affect the error at all: 
```javascript
newElement.find('span').each(function() {
        // This is needed because DAL will be duplicated otherwise
        $(this).remove();
    });
```

The main function in jquery seeming to remove DAL from the previous formsets is `$(selector).after(newElement);` commenting that line out seems to not remove DAL from the previous formset, however it then does not add the new formset either. So this won't work.

**Comment by jpic** [comment]
Interestingly, I did not find any JS change in-between 3.3.5 and 3.4.1.
@VenturaFranklin can you double-check it's still working for you on 3.3.5 ?
Would be great to know which version introduced the problem.

**Comment by VenturaFranklin** [comment]
Yes! I adjusted to 3.3.5 and it works, I pip install 3.4.1 and it doesn't work. 

Making no change in code literally reloading after installing the different versions.

**Comment by jpic** [comment]
That's interesting, is there any chance that you can reproduce in a new app inside test_project ?

**Comment by jpic** [comment]
Thanks for the demo @VenturaFranklin, the "django admin" way seems working, what do you suggest we do from here ?
